nomenklatura/enrich/openfigi.py

```diff
@@ -0,0 +1,77 @@
+import os
+import logging
+from typing import Any, Generator, Dict, List
+from urllib.parse import urljoin
+from followthemoney.util import make_entity_id
+from normality import slugify
+
+from nomenklatura.entity import CE
+from nomenklatura.dataset import DS
+from nomenklatura.cache import Cache
+from nomenklatura.enrich.common import Enricher, EnricherConfig
+
+log = logging.getLogger(__name__)
+
+URL = "https://api.openfigi.com/v3/search"
+
+
+class OpenFIGIEnricher(Enricher):
+    """Uses the `OpenFIGI` search API to look up FIGIs by company name."""
+
+    def __init__(self, dataset: DS, cache: Cache, config: EnricherConfig):
+        super().__init__(dataset, cache, config)
+
+        api_key = os.environ.get("OPENFIGI_API_KEY")
+        self.session.headers["X-OPENFIGI-APIKEY"] = api_key
+
+    def make_company_id(self, name):
+        return f"figi-co-{make_entity_id(name)}"
+
+    def make_security_id(self, figi):
+        return f"figi-id-{slugify(figi, sep='-')}"
```

Comment:
should these perhaps rather be something like `ofg-{hash}` for companies and `figi-{figi}` for figis?